### PR TITLE
test(logger): splitting existing tests to avoid pipeline failing for timeout

### LINKED ISSUE #5059: Test log flaky
https://github.com/fastify/fastify/actions/runs/6289773425/attempts/1?pr=5058

The `file option` test is flaky - we should refactor a bit the `test/serial/logger.1.test.js` and spit it into smaller files as we did before.

This should reduce the flakiness

### PR DESCRIPTION
**Pull request by giuliowaitforitdavide**
Closes #5059 

#### Checklist

- [X] run `npm run test` and `npm run benchmark`
- [X] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [X] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: test/logger/loggerTestUtils.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+const http = require('node:http')
```
Comment (Eomm) [inline test/logger/loggerTestUtils.js:1]:
```suggestion
'use strict'

const http = require('node:http')
```

File: test/logger/instantiation.test.js [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+    t.teardown(fastify.close.bind(fastify))
+
+    fastify.get('/foo', function (req, reply) {
+      t.ok(req.log)
+      req.log.info('log success')
+      reply.send({ hello: 'world' })
+    })
+
+    await fastify.ready()
+    await fastify.listen({ port: 0, host: localhost })
```
Comment (Eomm) [inline test/logger/instantiation.test.js:47]: minor optimization - we can remove the ready call

```suggestion
    await fastify.listen({ port: 0, host: localhost })
```

**Review by metcoder95 — APPROVED** [review]
Nice work, thanks for the support on this 🚀 

LGTM with @Eomm considerations

**Review by mcollina — APPROVED** [review]
lgtm

**Review by Uzlopak — APPROVED** [review]
LGTM

**Review by Eomm — APPROVED** [review]
3 runs in a row without flaky 🥹

Thanks!

**Pull request merged**